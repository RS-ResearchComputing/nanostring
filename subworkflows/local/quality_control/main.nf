//
// Perform quality control
//

include { NACHO_QC } from '../../../modules/nf-core/nacho/qc/main'

workflow QUALITY_CONTROL {
    take:
    counts      // channel: [ meta, [rcc files] ]
    samplesheet // channel: [ meta, path/to/samplesheet.csv ]

    main:
    NACHO_QC ( counts, samplesheet )

    emit:
    nacho_qc_multiqc_metrics = NACHO_QC.out.nacho_qc_png.map{it[1]}.mix(NACHO_QC.out.nacho_qc_txt.map{it[1]}) // channel: [ .png and .txt mqc files ]
    versions                 = NACHO_QC.out.versions                                                          // channel: [ versions.yml ]
}
