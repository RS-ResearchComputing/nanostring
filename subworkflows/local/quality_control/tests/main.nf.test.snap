{
    "test profile": {
        "content": [
            [
                "AVG_vs_BD_mqc.png",
                "AVG_vs_MED_mqc.png",
                "BD_mqc.png",
                "FOV_mqc.png",
                "HKF_mqc.png",
                "HK_mqc.png",
                "LOD_mqc.png",
                "Neg_mqc.png",
                "PCA1_vs_PCA2_mqc.png",
                "PCA_mqc.png",
                "PCAi_mqc.png",
                "POSF_vs_NEGF_mqc.png",
                "Pos_mqc.png",
                "Pos_vs_neg_mqc.png",
                "Posctrl_linearity_mqc.png",
                "hk_detected_mqc.txt",
                "normalized_qc_mqc.txt",
                "plot_normf_mqc.png"
            ],
            [
                "hk_detected_mqc.txt:md5,61209383acc2abf6fc3ea309b5a5e094",
                "normalized_qc_mqc.txt:md5,9a0b015a28094a17331b12b08898da8e"
            ],
            [
                "versions.yml:md5,70a6cd228c2f2620a4484f46566b82d8"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-14T20:29:29.43781046"
    }
}